sycl/test-e2e/InvokeSimd/Feature/invoke_simd_struct.cpp

```diff
@@ -53,58 +42,70 @@ using namespace sycl::ext::oneapi::experimental;
 namespace esimd = sycl::ext::intel::esimd;
 constexpr int VL = 16;
 
-struct multipliers {
-  int x;
-  int y;
+template <class T> struct solo {
+  T x;
+};
+
+template <class TY, class TX> struct duo {
+  TY y;
+  TX x;
 };
 
-/* Performs the addition of vectors A + B scaled by scalars->x and scalars->y.*/
-__attribute__((always_inline)) esimd::simd<int, VL>
-ESIMD_CALLEE(int *A, esimd::simd<int, VL> b, int i,
-             multipliers *scalars) SYCL_ESIMD_FUNCTION {
-  esimd::simd<int, VL> a;
+template <class TZ, class TY, class TX> struct trio {
+  TZ z;
+  TY y;
+  TX x;
+};
+
+template <class StructTy>
+__attribute__((always_inline)) esimd::simd<float, VL>
+ESIMD_CALLEE(float *A, int i, StructTy S) SYCL_ESIMD_FUNCTION {
+  esimd::simd<float, VL> a;
   a.copy_from(A + i);
-  // Add vectors A + B and scale them by x and y.
-  return a + b * scalars->x * scalars->y;
+  return a * S.x;
```

Comment:
Would you please sum x+y+z if StructTy is a 'trio' ? That would help to ensure that all fields are in their places and no padding errors happened.

Reply:
Done, also added one more case with function member of a structure being called.